code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -167,6 +178,10 @@
 
 /mob/living/simple_animal/hostile/headcrab/poison/AttackingTarget()
 	. = ..()
-	var/mob/living/carbon/C = target
-	if(C.can_inject(null, 0, "head", 0))
-		C.reagents.add_reagent("lsd", 5)
+	if(iscarbon(target) && target.reagents)
+		var/inject_target = pick("chest", "head")
+		var/mob/living/carbon/C = target
+		if(C.stunned || C.can_inject(null, 0, inject_target, 0))
```

Comment:
Unsure if you want to have an else clause for this condition, giving a different message indicating that it failed to pierce armor.
Also, unsure if this will result in 2 messages per attack given that you use . = ..() above.